dbms/src/Access/AllowedClientHosts.cpp

```diff
@@ -23,29 +24,62 @@ namespace ErrorCodes
 namespace
 {
     using IPAddress = Poco::Net::IPAddress;
+    using IPSubnet = AllowedClientHosts::IPSubnet;
+    const IPSubnet ALL_ADDRESSES{IPAddress{IPAddress::IPv6}, IPAddress{IPAddress::IPv6}};
 
-    const AllowedClientHosts::IPSubnet ALL_ADDRESSES = AllowedClientHosts::IPSubnet{IPAddress{IPAddress::IPv6}, IPAddress{IPAddress::IPv6}};
+    const IPAddress & getIPV6Loopback()
+    {
+        static const IPAddress ip("::1");
+        return ip;
+    }
+
+    bool isIPV4LoopbackMappedToIPV6(const IPAddress & ip)
+    {
+        static const IPAddress prefix("::ffff:127.0.0.0");
+        return (ip & IPAddress(104, IPAddress::IPv6)) == prefix;
```

Comment:
104?

Reply:
We have to reset the last 24 bits in `::ffff:127.x.y.z` to be able to check the first 128-24=104 bits.